expo/expo
# [metro-config] fix hermes internalbytecode symbolication error

# Why

when using hermes and the application throwing an unhandled promise rejection, our symbolication will show an error for file not found on windows.  
fixes #20305

# How

hermes builds promise polyfill in its internal bytecode. #20305 showing my working environment path because the versioned hermes was building on my machine. however, the error also happens on ios and maybe managed project. compared to bare react-native project, the root cause is from our `customizeFrame` metro config. this will make the InternalBytecode stack frame call to metro `/symbolicate` endpoint and get the error. this pr tries to add InternalBytecode to ignore frame list and prevent the error.

# Test Plan

tested on windows powershell

1. `yarn create expo-app -t blank@sdk-47 sdk47`
2. `cd sdk47`
3. edit App.js for throwing unhandled promise rejection:

```tsx
export default function App() {
  async function foo() {
    throw new Error("unhandled promise rejection");
  }

  useEffect(() => {
    foo();
  }, []);
 // ...
}
```

4. edit `node_modules/@expo/metro-config/build/ExpoMetroConfig.js` with the fix and check whether the error happens or not.

# Checklist

- n/a Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
